crates/sui-core/src/authority_client.rs

```diff
@@ -305,23 +306,30 @@ pub fn make_network_authority_clients_with_network_config(
     for (name, (_state, network_metadata)) in committee.validators() {
         let address = network_metadata.network_address.clone();
         let address = address.rewrite_udp_to_tcp();
-        // TODO: Enable TLS on this interface with below config, once support is rolled out to validators.
-        // let tls_config = network_metadata.network_public_key.as_ref().map(|key| {
-        //     sui_tls::create_rustls_client_config(
-        //         key.clone(),
-        //         sui_tls::SUI_VALIDATOR_SERVER_NAME.to_string(),
-        //         None,
-        //     )
-        // });
-        // TODO: Change below code to generate a SuiError if no valid TLS config is available.
-        let maybe_channel = network_config.connect_lazy(&address, None).map_err(|e| {
-            tracing::error!(
-                address = %address,
-                name = %name,
-                "unable to create authority client: {e}"
-            );
-            e.to_string().into()
-        });
+        let tls_config = network_metadata
+            .network_public_key
+            .as_ref()
+            .map(|key| {
+                sui_tls::create_rustls_client_config(
+                    key.clone(),
+                    sui_tls::SUI_VALIDATOR_SERVER_NAME.to_string(),
+                    None,
+                )
+            })
+            .ok_or(SuiError::from("network public key is not available"));
+        let maybe_channel = tls_config
+            .and_then(|tls_config| {
+                network_config
+                    .connect_lazy(&address, Some(tls_config))
```

Comment:
Do we want to make sure that the address provided uses https scheme?

Reply:
I think that if we do that we will break a bunch of validators probably?

Reply:
Does the address include http vs https, and do we need to force using https?

Reply:
Before this PR the indicated URL schema was ignored and we connected via HTTP. After this PR the indicated URL schema will continue to be ignored and we will connect via HTTPS.

Reply:
@bmwill i found our prior discussion on this here LOL
https://github.com/MystenLabs/sui/pull/19796#discussion_r1821767268

I just added `rewrite_http_to_https` as we (apparently) previously discussed 😅